homeassistant/components/http.py

```diff
@@ -201,11 +201,12 @@ def to_url(self, value):
     class DateValidator(BaseConverter):
         """Validate dates in urls."""
 
-        regex = r'\d{4}-\d{1,2}-\d{1,2}'
+        regex = r'\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d' \
```

Comment:
You can't repurpose a `DateValidator` as a datetime validator. Please create a new validator.